### ISSUE TITLE
Migrate to GitHub Actions?

### ISSUE THREAD

**Issue by conao3** [Silex/docker-emacs#73]
https://github.com/Silex/docker-emacs/issues/72#issuecomment-777229286

> We need to discuss architecture/hardware etc. I'm currently building the images on gitlab using an arm32/64 private server borrowed from packet.com and the usual gitlab workers for the i386/x64 images.

> Moving to purcell's nix would probably mean we use it to help build the arm versions of the nix packages and for the docker images we can use something simpler like travis.

**Comment by conao3** [comment]
I surprised @Silex use GitLab with arm32/64 private server.  Maybe we need to migrate to GitHubActions to make your project free and sustainable?
What reason to borrow arm32/64 server to build Docker image?

(I'm new at building Docker image.  Is there a problem with the intel server?)

**Comment by Silex** [comment]
Oh boy buckle up :-)

If you have the patience please read https://github.com/Silex/docker-emacs/pull/38

The TL;DR version of it is:

- Emacs requires ASLR to be disabled in order to build (see https://github.com/moby/moby/issues/22801) for versions < 27.1
- Some CI allows you to disable ASLR, but usually only for amd64/x86 variants, not for arm32v7/arm64v8 (maybe that changed).
- The crosscompiling route where you build the arm images on amd64:
  - Took ~90 minutes per images instead of 15 (some CI just timeout your jobs earlier than that, some others don't give enough free minutes per months).
  -  Failed with segfaults quite often, either at build time or when running Emacs. You have no idea how much I tried ot make it work 😭 
  
That's why I managed to get a server from packet.com (for free because they support open source) and I use gitlab because the orchestration is much more elaborated than the travis-ci one (see https://gitlab.com/Silex777/docker-emacs/-/pipelines/250539240 for an idea of how it works, each image depends on the previous one without the need to wait for the whole stage to finish).

If we could move to github actions I'd be pleased, it was not really mature back when I switched to gitlab. I'm not sure we can tho.

**Comment by Silex** [comment]
Note: if we manage to switch to @purcell nixos packages (by helping him only with the arm32/64 nix packages with our packet.com server), then all this ASLR bullshit can go away and we'd be able to use github actions with crossbuilding.

**Comment by Silex** [comment]
Alright, this is probably the next step.

**Comment by Silex** [comment]
It's done :tada:
